File: TarjetaEstilosSpring/src/main/java/pe/tarjetaestilos/models/Blog.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package pe.tarjetaestilos.models;

/**
 *
 * @author julito
 */

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.ZonedDateTime;
import java.util.ArrayList;
import java.util.List;
import org.hibernate.annotations.Where;


@Entity
@Table(name = "blog")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Where(clause = "deleted_at IS NULL")
public class Blog {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;
        
    @Column(columnDefinition = "TEXT")
    private String description;
    
    private String summary;

    @Column(name = "publication_date")
    private ZonedDateTime publicationDate;

    @Column(name = "reading_time")
    private String readingTime;

    @Column(name = "card_image")
    private String cardImage;

    @Column(name = "main_image")
    private String mainImage;
    
    @ManyToOne()
    @JoinColumn(name = "menu_id", nullable = false)
    @ToString.Exclude
    @JsonBackReference
    private Menu menu;
    
    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private ZonedDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private ZonedDateTime updatedAt;

    @Column(name = "deleted_at")
    private ZonedDateTime deletedAt;

    @OneToMany(mappedBy = "blog")
    @JsonManagedReference
    @ToString.Exclude
    private List<BlogCategory> blogCategories = new ArrayList<>();

}

File: src/models/Blog.ts
class Blog{    
    title:string|null;
    description:string|null;
    summary:string|null;
    publication_date:string|null;
    reading_timed:string|null;
    card_image:string|null;
    main_image:string|null;
    created_at:Date|null;
    updated_at:Date|null;
    deleted_at:Date|null;
    startDate:string|null;
    endDate:string|null;
    category_id:number[]|null;
    id:number|null;
    menu_id:number|null;

    constructor(title:string|null, description:string|null, summary:string|null, publication_date: string|null, reading_timed:string|null,  
        card_image:string|null,
        main_image:string|null,
        created_at:Date|null, updated_at:Date|null, deleted_at:Date|null, startDate:string|null=null, endDate:string|null=null,
        category_id:number[]|null=null, id:number|null=null,menu_id:number|null=null
    ){
        this.title = title;
        this.description = description;
        this.summary = summary;
        this.publication_date = publication_date;
        this.reading_timed = reading_timed;
        this.card_image = card_image;
        this.main_image = main_image;
        this.created_at = created_at;
        this.updated_at = updated_at;
        this.deleted_at = deleted_at;
        this.startDate = startDate;
        this.endDate = endDate;
        this.category_id = category_id;
        this.id = id;
        this.menu_id = menu_id;
    }
};
export default Blog;